internal/backend/sessionsgrpcsvc/svc.go

```diff
@@ -78,6 +75,37 @@ func (s *server) Start(ctx context.Context, req *connect.Request[sessionsv1.Star
 	return connect.NewResponse(&sessionsv1.StartResponse{SessionId: uid.String()}), nil
 }
 
+func decodeNestedJSON(input string) (any, error) {
+	var result any
+	d := json.NewDecoder(strings.NewReader(input))
+	d.UseNumber()
+
+	if err := d.Decode(&result); err != nil {
+		return nil, err
+	}
+
+	switch v := result.(type) {
+	case map[string]any:
+		for key, value := range v {
+			if str, ok := value.(string); ok {
+				if decoded, err := decodeNestedJSON(str); err == nil {
```

Comment:
I understand the idea, but I'm not sure it's the right approach. We'll need to discuss it as we don't know if the user is intending for this to be decoded. I think that the sender of the data can put it explicitly as json.

Reply:
I think we need to use ``map[string]interface{}{}`` here instead of ``map[string][string]``: https://github.com/autokitteh/autokitteh/blob/efc48f56d7356718a7df8d2ca86c112c5dc6d1cf/proto/gen/go/autokitteh/sessions/v1/svc.pb.go#L32

As explained here: https://www.digitalocean.com/community/tutorials/how-to-use-json-in-go

Reply:
i went over the code again and now i think your original approach is ok since this is the `JsonInputs` field and not what that is supplied in the Session itself.

That said, the implementation you suggested is insufficient - you might get types that are not []any or map[string]any, such as map[int]string or []int, or []string, which this will not catch. One need to use the `reflect` module here to work these over. I can try tackling this.

Reply:
looks like it should work! according to https://pkg.go.dev/encoding/json#Unmarshal, it will result in either []any or map[string]any. cool.